ros_ws/src/controller/src/onnx_handler.cpp

```diff
@@ -0,0 +1,72 @@
+#include "../include/onnx_handler.hpp"
+
+OnnxHandler::OnnxHandler(const std::string _path, const int _num_inputs, const int _num_ouputs)
+    : path(_path), num_inputs(_num_inputs), num_outputs(_num_ouputs){
+
+  init_onnx_session();
+
+  // shape specifies the dimensions of the input/output sensor
+  input_shape = {num_inputs};
+  output_shape = {num_outputs};
+
+  input_buffer = std::vector<float>(_num_inputs);
+  output_buffer = std::vector<float>(_num_ouputs);
+
+  auto memory_info = Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
+
+  input_tensor = Ort::Value::CreateTensor<float>(
+      memory_info, input_buffer.data(), input_buffer.size(), input_shape.data(),
+      input_shape.size());
+
+  output_tensor = Ort::Value::CreateTensor<float>(
+      memory_info, output_buffer.data(), output_buffer.size(), output_shape.data(),
+      output_shape.size());
+
+  std::cout << "Memory information : \n"
+            << memory_info << std::endl;
+
+  std::cout << "Input and output buffers have been initiliazed!\n"
+            << "Address of the controller :\t" << memory_info << std::endl
+            << "Input tensor " << input_tensor << "\t tensor size: " << input_buffer.size() << std::endl
+            << "Output tensor " << output_tensor << "\t tensor size: " << output_buffer.size() << std::endl;
```

Comment:
`printf`?

Reply:
The stream is more neat to be able to print on the console and write to a file at the same time.